src/Base/Exception.h

```diff
@@ -33,6 +33,13 @@
 #include "FileInfo.h"
 #include "BaseClass.h"
 
+/* MACROS FOR THROWING EXCEPTIONS */
+
+#define THROW(exception) throw Base::exception(__FILE__,__LINE__,__PRETTY_FUNCTION__);
+#define THROWM(exception, message) throw Base::exception(message,__FILE__,__LINE__,__PRETTY_FUNCTION__);

```

Comment:
It is intended.
Not any exception does implement a constructor as required to make it work. Base::Exception and derived exceptions do because of inheritance.

Now, if you tell me that you are interested in using such macros elsewhere in the same form (with just those parameters), we might think of putting the macros elsewhere (so that you can include them without including Base::Exception) and then remove the Base::

I